File: frmSignUp.php
<?php
session_start();
require_once "config.php";

$username = "";
$password = "";
$contact = "";
$error = "";
$success = "";

if ($_SERVER["REQUEST_METHOD"] == "POST" && isset($_POST["signup"])) {
    $username = trim($_POST["username"]);
    $password = trim($_POST["password"]);
    $contact = trim($_POST["contact"]);
    $dateCreated = date("m/d/Y");

    // Validation
    if (strlen($password) < 6) {
        $error = "Password must be at least 6 characters.";
    } else {
        // Check if username exists
        $checkSql = "SELECT * FROM tbluser WHERE username = ?";
        if ($stmt = mysqli_prepare($link, $checkSql)) {
            mysqli_stmt_bind_param($stmt, "s", $username);
            mysqli_stmt_execute($stmt);
            mysqli_stmt_store_result($stmt);
            if (mysqli_stmt_num_rows($stmt) > 0) {
                $error = "Username already exists.";
            } else {
                // Insert into tbluser
                $insertSql = "INSERT INTO tbluser (username, password, usertype, status, contact, DateCreated, CreatedBy)
                              VALUES (?, ?, 'CUSTOMER', 'ACTIVE', ?, ?, ?)";
                if ($stmtInsert = mysqli_prepare($link, $insertSql)) {
                    mysqli_stmt_bind_param($stmtInsert, "sssss", $username, $password, $contact, $dateCreated, $username);
                    if (mysqli_stmt_execute($stmtInsert)) {
                        // Insert into tbllogs
                        $logSql = "INSERT INTO tbllogs (datelog, timelog, action, module, ID, performedby)
                                   VALUES (?, ?, 'Add', 'Accounts Management', 'CUSTOMER', ?)";
                        $datelog = date("m/d/Y");
                        $timelog = date("h:i A");
                        if ($stmtLog = mysqli_prepare($link, $logSql)) {
                            mysqli_stmt_bind_param($stmtLog, "sss", $datelog, $timelog, $username);
                            mysqli_stmt_execute($stmtLog);
                            mysqli_stmt_close($stmtLog);
                        }
                        // Set success message
                        $success = "Account created successfully!";
                    } else {
                        $error = "Something went wrong while registering. Try again.";
                    }
                    mysqli_stmt_close($stmtInsert);
                }
            }
            mysqli_stmt_close($stmt);
        }
    }
}

mysqli_close($link);
?>

<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>Armando - Sign Up</title>
  <style>
    body {
      background-color: #cceaff;
      font-family: 'Arial Rounded MT Bold', sans-serif;
      display: flex;
      justify-content: center;
      align-items: center;
      height: 100vh;
      margin: 0;
    }
    .container {
      text-align: center;
    }
    h1 {
      font-size: 5rem;
      color: #8e60d4;
      text-shadow: 0px 4px 6px rgba(0, 0, 0, 0.1);
      margin-bottom: 0.3rem;
    }
    .subtitle {
      font-size: 1.5rem;
      letter-spacing: 2px;
      color: #000;
      margin-bottom: 2rem;
    }
    input[type="text"],
    input[type="password"] {
      width: 300px;
      padding: 15px;
      border: none;
      border-radius: 30px;
      margin: 10px 0;
      font-size: 1rem;
      background-color: #fff;
      box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.1);
    }
    .btn {
      padding: 15px 40px;
      margin: 10px;
      border: none;
      border-radius: 30px;
      font-size: 1.1rem;
      background-color: #6246d1;
      color: white;
      cursor: pointer;
      transition: background-color 0.3s ease;
    }
    .btn:hover {
      background-color: #5139af;
    }
    .login-text {
      margin-top: 10px;
      font-size: 0.95rem;
    }
    .login-text a {
      color: #6246d1;
      text-decoration: underline;
      cursor: pointer;
    }
    .message {
      color: red;
      margin-top: 10px;
    }
    .message.success {
      color: green;
    }
    /* Pop-up notification styles */
    #popup-notification {
      position: fixed;
      top: 0;
      left: 50%;
      transform: translateX(-50%);
      background-color: #4caf50;
      color: white;
      padding: 15px;
      font-size: 1.2rem;
      border-radius: 5px;
      display: none;
      z-index: 9999;
    }
    .popup-btn {
      background-color: #3e8e41;
      padding: 10px;
      border-radius: 5px;
      border: none;
      cursor: pointer;
      color: white;
      font-size: 1rem;
    }
  </style>
</head>
<body>
  <div class="container">
    <h1>Armando</h1>
    <div class="subtitle">BIKE SHOP, PARTS & SERVICES</div>

    <?php if (!empty($error)) : ?>
      <div class="message"><?php echo $error; ?></div>
    <?php endif; ?>

    <?php if (!empty($success)) : ?>
      <div id="popup-notification">
        <?php echo $success; ?>
        <br>
        <button class="popup-btn" onclick="redirectToLogin()">OK</button>
      </div>
      <script>
        // Show the pop-up notification
        document.getElementById('popup-notification').style.display = 'block';

        // Function to redirect to login page
        function redirectToLogin() {
          window.location.href = "frmLogin.php";
        }
      </script>
    <?php endif; ?>

    <form method="post" action="">
      <input type="text" name="username" placeholder="Username" value="<?php echo htmlspecialchars($username); ?>" required><br>
      <input type="password" name="password" placeholder="Password" value="<?php echo htmlspecialchars($password); ?>" required><br>
      <input type="text" name="contact" placeholder="Contact Number" value="<?php echo htmlspecialchars($contact); ?>" required><br>
      <button type="submit" name="signup" class="btn">Sign Up</button>
    </form>

    <div class="login-text">
      Already have an account? <a href="frmLogin.php">Login</a>
    </div>
  </div>
</body>
</html>

File: frmSignUp.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Drawing.Imaging;
using System.Linq;
using System.Security.Principal;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using static System.Windows.Forms.VisualStyles.VisualStyleElement.StartPanel;

namespace CS311A2024_DATABASE
{
    public partial class frmSignUp : Form
    {
        public frmSignUp()
        {
            InitializeComponent();
        }
        private int errorcount;
        Class1 newuser = new Class1("127.0.0.1", "cs311a2024", "lee", "12345");

        private void validateForm()
        {
            errorcount = 0;
            errorProvider1.Clear();
            if (string.IsNullOrEmpty(txtusername.Text))
            {
                errorProvider1.SetError(txtusername, "Input is empty");
                errorcount++;
            }
            if (string.IsNullOrEmpty(txtpassword.Text))
            {
                errorProvider1.SetError(txtpassword, "Input is empty");
                errorcount++;
            }
            if (txtpassword.Text.Length <= 3)
            {
                errorProvider1.SetError(txtpassword, "Pasword must be more than 4 characters");
                    errorcount++;
            }
            if (string.IsNullOrEmpty(txtreenter.Text))
            {
                errorProvider1.SetError(txtreenter, "Input is empty");
                errorcount++;
            }
            if (string.IsNullOrEmpty(txtcontact.Text))
            {
                errorProvider1.SetError(txtcontact, "Input is empty");
                errorcount++;
            }
            if (txtreenter.Text != txtpassword.Text)
            {
                errorProvider1.SetError(txtreenter, "Password is not the same");
                errorcount++;
            }
            try
            {
                DataTable dt = newuser.GetData("SELECT * FROM tbluser WHERE username = '" + txtusername.Text + "'");
                if (dt.Rows.Count > 0)
                {
                    errorProvider1.SetError(txtusername, "Username is already in use.");
                    errorcount++;
                }
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message, "Error on validating existing username", MessageBoxButtons.OK, MessageBoxIcon.Error);
            }
        }

        private void btncancel_Click(object sender, EventArgs e)
        {
            frmLogin frmLogin = new frmLogin();
            frmLogin.Show();
            this.Hide();
        }

        private void btnsignup_Click(object sender, EventArgs e)
        {
            validateForm();
            if (errorcount == 0)
            {
                DialogResult dr = MessageBox.Show("Are you sure you want to add this account?", "Confirmation", MessageBoxButtons.YesNo, MessageBoxIcon.Question);
                if (dr == DialogResult.Yes)
                {
                    try
                    {
                        newuser.executeSQL("INSERT INTO tbluser (username, password, usertype, status, email, birthdate, contact, CreatedBy, DateCreated) " +
                            "VALUES ('" + txtusername.Text + "', '" + txtpassword.Text + "', 'CUSTOMER', 'ACTIVE', 'Insert Email', 'Insert Birthdate MM/DD/YYYY', '" + txtcontact.Text + "', 'NONE', '" + DateTime.Now.ToShortDateString() + "')");

                        if (newuser.rowAffected > 0)
                        {
                            newuser.executeSQL("INSERT INTO tbllogs (datelog, timelog, action, module, ID, performedby) " +
                                "VALUES ('" + DateTime.Now.ToShortDateString() + "', '" + DateTime.Now.ToShortTimeString() + "', 'Add', 'Accounts Management', 'CUSTOMER', 'CUSTOMER')");

                            MessageBox.Show("Account Added", "Message", MessageBoxButtons.OK, MessageBoxIcon.Information);
                            this.Close();
                            frmLogin frmLogin = new frmLogin();
                            frmLogin.Show();
                        }
                    }
                    catch (Exception ex)
                    {
                        MessageBox.Show(ex.Message, "Error on save", MessageBoxButtons.OK, MessageBoxIcon.Error);
                    }
                }
            }
        }
    }
}
